src/IBM.Watson.TextToSpeech.v1/Websockets/AWebSocketClient.cs

```diff
@@ -0,0 +1,188 @@
+
+            return this;
+        }
+        public AWebSocketClient WithAuthentication(string userName, string password)
+        {
+            BaseClient.Options.SetRequestHeader("Authorization", "Basic " + Convert.ToBase64String(Encoding.ASCII.GetBytes(userName + ":" + password)));
+            return this;
+        }
+        public AWebSocketClient WithHeader(string headerName, string headerValue)
+        {
+            this.BaseClient.Options.SetRequestHeader(headerName, headerValue);
+            return this;
+        }
+
+        public abstract void Send(FileStream file, string openingMessage);
+
+        public abstract void Send(string request, string accept, string[] timings, string openingMessage);
+
+        protected async Task SendAudio(FileStream file)
+        {
+            byte[] b = new byte[1024];
+            while (file.Read(b, 0, b.Length) > 0)
+            {
+                await BaseClient.SendAsync(new ArraySegment<byte>(b), WebSocketMessageType.Binary, true, CancellationToken.None);
+            }
+            await BaseClient.SendAsync(stopMessage, WebSocketMessageType.Text, true, CancellationToken.None);
+        }
+
+        protected async Task SendText(string message)
+        {
+            var messageBuffer = Encoding.UTF8.GetBytes(message);
+            var messagesCount = (int)Math.Ceiling((double)messageBuffer.Length / SendChunkSize);
+
+            for (var i = 0; i < messagesCount; i++)
+            {
+                var offset = (SendChunkSize * i);
+                var count = SendChunkSize;
+                var lastMessage = ((i + 1) == messagesCount);
+
+                if ((count * (i + 1)) > messageBuffer.Length)
+                {
+                    count = messageBuffer.Length - offset;
+                }
+                await BaseClient.SendAsync(new ArraySegment<byte>(messageBuffer, offset, count), WebSocketMessageType.Text, lastMessage, CancellationToken.None);
+            }
+        }
+
+        protected async Task HandleResults()
+        {
+            var buffer = new byte[1024 * 16 * 4];
+            var audioStream = new List<byte>();
+
+            while (true)
+            {
+
+                var segment = new ArraySegment<byte>(buffer);
+
+                var result = await BaseClient.ReceiveAsync(segment, CancellationToken.None);
+
+                if (result.MessageType == WebSocketMessageType.Close)
+                {
+                    return;
+                }
+                //58290
+                int count = result.Count;
+                while (!result.EndOfMessage)
+                {
+                    if (count >= buffer.Length)
+                    {
+                        if (BaseClient.State == WebSocketState.Open)
+                        {
+                            await BaseClient.CloseAsync(WebSocketCloseStatus.InvalidPayloadData, "That's too long", CancellationToken.None);
```

Comment:
Does the user see this message? If so, can you put a more descriptive message here?

Reply:
I need to test this out with a long text request to see if it gets triggered

Reply:
ok, so we should have a more descriptive message here. From https://cloud.ibm.com/apidocs/text-to-speech#synthesize-audio-websockets-

Says you can provide a maximum of 5kb of text

Reply:
So this is never reached, I updated the code. The situation here is we either do it like my updated way in which it calls the `onError` callback to show the error, or the error message is automatically sent to the `onMessage` callback by the service.